cyipopt/scipy_interface.py

```diff
@@ -324,11 +324,77 @@ def minimize_ipopt(fun,
+
+    This function can be used to solve general nonlinear programming problems
+    of the form:
+
+    .. math::
+
+       \min_ {x \in R^n} f(x)
+
+    subject to
+
+    .. math::
+
+       g_L \leq g(x) \leq g_U
+
+       x_L \leq  x  \leq x_U
+
+    Where :math:`x` are the optimization variables, :math:`f(x)` is the
+    objective function, :math:`g(x)` are the general nonlinear constraints,
+    and :math:`x_L` and :math:`x_U` are the upper and lower bounds
+    (respectively) on the decision variables. The constraints, :math:`g(x)`,
+    have lower and upper bounds :math:`g_L` and :math:`g_U`. Note that equality
+    constraints can be specified by setting :math:`g^i_L = g^i_U`.
+
+    Parameters
+    ----------
+    fun : callable
+        The objective function to be minimized: ``fun(x, *args, **kwargs) ->
+        float``.
+    x0 : array-like, shape(n, )
+        Initial guess. Array of real elements of shape (n,),
+        where ``n`` is the number of independent variables.
+    args : tuple, optional
+        Extra arguments passed to the objective function and its
+        derivatives (``fun``, ``jac``, and ``hess``).
+    kwargs : dictionary, optional
+        Extra keyword arguments passed to the objective function and its
+        derivatives (``fun``, ``jac``, ``hess``).
+    method : str, optional
+        This parameter is ignored. `minimize_ipopt` always uses Ipopt; use
+        `scipy.optimize.minimize` directly for other methods.
+    jac : callable, optional
+        The Jacobian of the objective function: ``jac(x, *args, **kwargs) ->
+        ndarray, shape(n, )``. If ``None``, SciPy's ``approx_fprime`` is used.
+    hess : callable, optional
+        If ``None``, the Hessian is computed using IPOPT's numerical methods.
+        Explicitly defined Hessians are not yet supported via this interface.
+    hessp : callable, optional
+        If ``None``, the Hessian is computed using IPOPT's numerical methods.
+        Explicitly defined Hessians are not yet supported via this interface.
+    bounds :  sequence, shape(n, ), optional
+        Sequence of ``(min, max)`` pairs for each element in `x`. Use ``None``
+        to specify no bound.
+    constraints : {Constraint, dict}, optional
+        See `scipy.optimize.minimize` for more information. Note that the
+        Jacobian of each constraint corresponds to the ``'jac'`` key and must
+        be a callable function with signature
+        ``jac(x) -> {ndarray, coo_array}``. If the constraint's
+        value of ``'jac'`` is ``True``, the constraint function ``fun`` must
+        return a tuple ``(con_val, con_jac)`` consisting of the evaluated
+        constraint ``con_val`` and the evaluated Jacobian ``con_jac``.
+    tol : float, optional (default=1e-8)
+        The desired relative convergence tolerance, passed as an option to
+        Ipopt. See [https://coin-or.github.io/Ipopt/OPTIONS.html] for details.
```

Comment:
If desired, I could add Ipopt documentation as a reference and make this
```suggestion
        Ipopt. See [1]_ for details.
```